kernel/nano_core/build.rs

```diff
@@ -79,30 +78,33 @@ fn emit_built_rs_file() {
     let mut custom_cfgs_str = String::new();
 
     for (k, v) in std::env::vars() {
-        if k.starts_with("CARGO_CFG_") && !NON_CUSTOM_CFGS.contains(&k.as_str()) {
-            let key = k[CARGO_CFG_PREFIX.len()..].to_lowercase();
-            custom_cfgs = format!("{custom_cfgs}(\"{key}\", \"{v}\"), ");
-
-            custom_cfgs_str.push_str(&key);
-            if !v.is_empty() {
-                write!(custom_cfgs_str, "=\"{v}\"").expect("Failed to write to custom_cfgs_str");
+        if let Some(key) = k.strip_prefix(CARGO_CFG_PREFIX) {
+            if !NON_CUSTOM_CFGS.contains(&key) {
+                let key = key.to_lowercase();
+                custom_cfgs = format!("{custom_cfgs}(\"{key}\", \"{v}\"), ");
+                custom_cfgs_str.push_str(&key);
+                if !v.is_empty() {
+                    write!(custom_cfgs_str, "=\"{v}\"").expect("Failed to write to custom_cfgs_str");
+                }
+                custom_cfgs_str.push(' ');
+    
+                num_custom_cfgs += 1;
             }
-            custom_cfgs_str.push(' ');
-
-            num_custom_cfgs += 1;
         }
     }
 
     // Append all of the custom cfg values to the built.rs file as an array.
     write!(
         &mut built_file,
-        "#[allow(dead_code)]\npub const CUSTOM_CFG: [(&str, &str); {num_custom_cfgs}] = [{custom_cfgs}];\n",
+        "#[allow(dead_code)]\n
```

Comment:
you need to append another `\` at the end of the line (after the `\n`) in order to keep the spacing alignment correct.

Reply:
I somehow thought normal rust strings don't care about new lines and only raw strings do care, oops